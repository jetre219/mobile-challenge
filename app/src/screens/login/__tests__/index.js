import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { LoginScreen } from '..';

let homeScreen;

const login = jest.fn();

describe('Home screen', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    homeScreen = render(<LoginScreen login={login} />);
  });

  it('renders correctly', () => {
    expect(homeScreen.toJSON()).toMatchSnapshot();
  });

  describe('when pressing login button', () => {
    beforeEach(() => {
      fireEvent.press(homeScreen.getByText('Login'));
    });

    it('calls the login function', () => {
      expect(login).toHaveBeenCalled();
    });
  });
});
