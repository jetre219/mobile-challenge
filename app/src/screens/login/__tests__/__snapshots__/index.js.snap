// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "overflow": "visible",
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 22,
              "marginHorizontal": 8,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
